table/tables/tables.go

```diff
@@ -509,13 +511,31 @@ func (t *tableCommon) AddRecord(ctx sessionctx.Context, r []types.Datum, opts ..
 		if !t.canSkip(col, value) {
 			colIDs = append(colIDs, col.ID)
 			row = append(row, value)
+			continue
+		}
+		// Fill column size for statistics.
+		if col.IsPKHandleColumn(t.Meta()) {
+			// The size of primary key column is 8 byte.
+			colSize[col.ID] = 8
+			continue
+		}
+		if col.IsGenerated() && !col.GeneratedStored {
+			// Lazy-initialization buffer.
+			if encodedCol == nil {
+				encodedCol = make([]byte, 0, 16)
+			}
+			encodedCol, err = tablecodec.EncodeValue(sessVars.StmtCtx, encodedCol[:0], value)
+			if err != nil {
+				continue
```

Comment:
I'm not sure if it is safe to `continue` here. @eurekaka

Reply:
I thinks it's fine, then this column would not have delta for column size.